javascript/app-framework/src/components/editors/TableBundleEditor/TableBundleEditor.tsx

```diff
@@ -2,20 +2,162 @@
  * Copyright (c) Microsoft. All rights reserved.
  * Licensed under the MIT license. See LICENSE file in the project.
  */
-import { memo } from 'react'
-import { useHelpOnMount } from '../../../hooks/useHelpOnMount.js'
-
-import { JsonModeEditor } from '../JsonModeEditor/JsonModeEditor.js'
-import { BundleEditor } from './BundleEditor.js'
-import type { TableBundleEditorProps } from './TableBundleEditor.types.js'
-
-export const TableBundleEditor: React.FC<TableBundleEditorProps> = memo(
-	function TableBundleEditor({ resource }) {
-		useHelpOnMount('resources.tablebundle.index')
-		return (
-			<JsonModeEditor resource={resource}>
-				<BundleEditor resource={resource} />
-			</JsonModeEditor>
+import {
+	ArqueroDetailsList,
+	ArqueroTableHeader,
+	DisplayOrder,
+	StepList,
+	TableCommands,
+} from '@datashaper/react'
+import { KnownProfile } from '@datashaper/schema'
+import { Workflow } from '@datashaper/workflow'
+import type { TableBundle } from '@datashaper/workflow'
+import { ToolPanel } from '@essex/components'
+import { CommandBar } from '@fluentui/react'
+import { useObservableState } from 'observable-hooks'
+import { memo, useCallback, useMemo, useState } from 'react'
+import type { PluginComponentProps } from '../../../types.js'
+
+import { useToolPanelExpandCollapse } from '../hooks.js'
+import {
+	useTableHeaderColors,
+	useTableHeaderStyles,
+	useToolPanelStyles,
+} from '../styles.js'
+import {
+	useColumnState,
+	useOnCreateStep,
+	useOnDeleteStep,
+	useOnSaveStep,
+	useSelectedTable,
+	useTableCommandProps,
+} from './TableBundleEditor.hooks.js'
+import { Container, DetailsListContainer } from './TableBundleEditor.styles.js'
+
+export const TableBundleEditor: React.FC<PluginComponentProps<TableBundle>> =
+	memo(function TableBundleEditor({ resource }) {
+		const [workflow, isWorkflowAttached] = useMemo<[Workflow, boolean]>(() => {
+			const result = resource
```

Comment:
`result` here is confusing me with `result` in defaultWorkflow. Maybe we call this variable `attachedWorkflow'?